src/hotspot/cpu/x86/frame_x86.cpp

```diff
@@ -253,7 +253,11 @@ bool frame::safe_for_sender(JavaThread *thread) {
 
   // Will the pc we fetch be non-zero (which we'll find at the oldest frame)
 
-  if ( (address) this->fp()[return_addr_offset] == NULL) return false;
+  // Will the pc we fetch be non-zero (which we'll find at the oldest frame) and readable
+
+  if (!thread->is_in_usable_stack((address)this->fp())) {
+    return false;
+  }
```

Comment:
This looks like a paste error - you've deleted a line that is needed and now you have two comment lines.